src/components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useIsApprovalNeeded.tsx

```diff
@@ -0,0 +1,60 @@
+import type { AccountId } from '@shapeshiftoss/caip'
+import { useMemo } from 'react'
+import { useAllowance } from 'components/MultiHopTrade/components/MultiHopTradeConfirm/hooks/useAllowance'
+import { bn } from 'lib/bignumber/bignumber'
+import type { TradeQuoteStep } from 'lib/swapper/types'
+import { assertUnreachable } from 'lib/utils'
+
+import { GetAllowanceErr } from '../../../hooks/useAllowanceApproval/helpers'
+
+export type UseIsApprovalNeededProps = {
+  watch: boolean
+  tradeQuoteStep: TradeQuoteStep | undefined
+  sellAssetAccountId: AccountId | undefined
+}
+
+export const useIsApprovalNeeded = ({
+  watch,
+  tradeQuoteStep,
+  sellAssetAccountId,
+}: UseIsApprovalNeededProps) => {
+  const { isLoading, data } = useAllowance({
+    sellAssetAccountId,
+    tradeQuoteStep,
+    watch,
+    enabled: true,
```

Comment:
No need for this since it's not programmatic and `useAllowance` itself already has its enabled checks

Reply:
90b8d200ec3a4c490d0be04ffbb0276c4ada23d2

Reply:
ty ser